Add thermal cutout for Tycho load resistors

martinling opened the pull request:
This PR adds a thermal cutout switch to the Tycho layout, placed between the load resistors in order to cut power to them if they heat the board excessively. This should not occur in normal usage but could happen if they are left switched on as a result of a software fault.

The footprint will fit any [Bourns SC](https://www.bourns.com/docs/product-datasheets/sc.pdf) part. Available cutoff temperatures are 72 to 85°C; I expect that any of these options would suffice in practice. I have selected the 85°C part which is the most readily available.

To maximise heat transfer to the package, the layout positions the cutout between the two load resistors, with its body sitting on a GND strip which runs between their heat sink pads.

![image](https://github.com/greatscottgadgets/tycho/assets/673823/26f8e4e6-af74-471e-9e4b-33a757ae67c4)

Testing will be required to verify that this solution provides an acceptable cutoff behaviour in practice.

martinling committed:
Add footprint for Bourns SC thermal cutout.

martinling committed:
Add thermal cutout for load resistors.

martinling committed:
Update DATE and bump VERSION to r2.0.0+.

mossmann reviewed (approved):
This looks great! I'm fine with merging this as-is, but I did spot some tiny improvements that could be made:
- specify Manufacturer for SW3
- correct F.Fab layer conflict between SW3 and R48
- add component outline to SW3 F.Fab layer

martinling committed:
Revise PWR163 footprint.

- Use centroid of main package body as footprint origin.
- Add F.Fab outline.
- Add F.Courtyard boundary.
- Add F.Silkscreen markings.

martinling commented:
@mossmann I've made those changes and some further ones.

Given the proximity of SW3 to the resistors I have updated both footprints to include the mechanical outline, silkscreen and courtyard, and based on this I have spaced them a bit further apart.

The pad sizes for the resistors are unchanged from the tested footprint, and these correspond to the recommendation in the datasheet. However I have added the additional layers, moved the origin to the centroid of the package body, and adjusted the paste mask to better correspond to the actual pad shape on the package.

I've also rotated SW3 the other way so that the base terminal is connected to the load side rather than the arm terminal. I think this orientation should give better heat transfer as the load side is more directly connected to the resistors.

![image](https://github.com/greatscottgadgets/tycho/assets/673823/dea10d05-c0cb-43a8-ac2d-7d5807067f12)
